internal/translation/deployment/deployment_test.go

```diff
@@ -1,3 +1,4 @@
+//nolint:dupl
```

Comment:
follow up: The need for adding these usually hides a bigger problem with out code or even its design. I suggest to create a ticket to review these usages.

Reply:
Personally I think that this is fine in tests, since we end up with a lot of boilerplate and repeated code with very small differences (especially with the mocks, which is what's happening here). Would be interested what other people think about this.

To be clear; I do think we should be avoiding using nolint:dupl (or nolint for anything) in our 'production' code.

Reply:
I do agree it might not always make sense to remove this in test code. Some times I think it does, [see this recent example|https://github.com/mongodb/mongodb-atlas-kubernetes/pull/2035/files], we were replicating all test cases just because the base "container" type was different, and we were going to replicate even more going forward.

Other times, it might not pay off. But I think it does make sense to sit back and analyse case by case **as a follow up**.

Reply:
I am on the same page with @roothorp on this test. Setting up those mocks looks similar from the linter perspective but they vary in many details. We had to introduce per-test setups as they were indeed varying. Introducing helper functions would make things very messy as it would need to take many parameters on the things that change.

An alternative could be though to create a prototypical object and just change it for the differences for each test but this is loosing transparency as I learned from the past as the reviewer does not grasp the whole object.